pkg/network/usm/postgres_monitor_test.go

```diff
@@ -112,6 +113,8 @@ func (s *postgresProtocolParsingSuite) TestLoadPostgresBinary() {
 
 func (s *postgresProtocolParsingSuite) TestDecoding() {
 	t := s.T()
+	// USMO-557
+	flake.Mark(t)
```

Comment:
The test will be [skipped](https://github.com/DataDog/datadog-agent/blob/5cabbd505727234b911b6729a4e4e691f8f33150/pkg/util/testutil/flake/flake.go#L28C5-L31) if `shouldSkipFlake` returns true

`shouldSkipFlake` returns true either if `skip-flake` flag was set, or `GO_TEST_SKIP_FLAKE` env was set to true.

This test runs under kmt, and `skip-flake` is being used only in `gotest.py` so there is no option to set this flag for kmt.

`GO_TEST_SKIP_FLAKE` is by default set to `true`, but if the branch is main it is being set to false.
so the test will run on main, and it will not solve the original issue you're complaining about

What did I miss?

Reply:
I thought that KMT implemented the flakes washer logic implemented in other go tests, where we post process tests within a job and allow to fail tests that are known flakes. Marking a test as flake resolves the impact on `main` as the job is safe, while the test keeps reporting the failures.

By your words it looks like this is not the case.

@brycekahle could KMT implement the logic used by the flake marker with other go test, preventing failing kmt job on known flakes failures?

Reply:
This is the PR that introduced the test washer for KMT https://github.com/DataDog/datadog-agent/pull/26793

The implementation respects what we implemented on invoke test side. Maybe discussing this online would help better understanding how the flake marker works, and what are the consequences on `main/dev branches`

Reply:
In other words

> shouldSkipFlake returns true either if skip-flake flag was set, or GO_TEST_SKIP_FLAKE env was set to true.
This test runs under kmt, and skip-flake is being used only in gotest.py so there is no option to set this flag for kmt.
GO_TEST_SKIP_FLAKE is by default set to true, but if the branch is main it is being set to false.
so the test will run on main

This is correct

> and it will not solve the original issue you're complaining about

This is not exact. We implemented the TestWasher, a post processing that allows failures for any marked tests. The test keeps reporting its result to Test Visibility, but the job and thus the pipeline does not fail.